### PR TITLE
Added commenting on posts and comments

### LINKED ISSUE #18: As a logged in user,  I would like to be able to comment on a post

### PR DESCRIPTION
**Pull request by rhysyngsun**
#### What are the relevant tickets?
Fixes #18 

#### What's this PR do?
This adds UI to add comments  in reply to posts and other comments.

#### How should this be manually tested?
Verify you can comment on a post and comment.

### CODE REVIEW

File: static/js/reducers/comments_test.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  })
+
+  it("should let you reply to a post", () => {
+    const { requestType, successType } = actions.comments.post
+    return listenForActions(
+      [actions.comments.get.successType, actions.comments.get.requestType, requestType, successType],
+      () => {
+        store.dispatch(actions.comments.get(post.id))
+        store.dispatch(actions.comments.post(post.id, "comment text"))
+      },
```
Comment (noisecapella) [inline static/js/reducers/comments_test.js]: The below code isn't getting run, did you mean to do `).then(state => ...` here?

File: static/js/components/CreateCommentForm_test.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        assert.equal(wrapper.find("a").text(), "Reply")
+      })
+
+      it("should show and empty when reply is clicked", () => {
+        return helper.listenForActions([
+          forms.FORM_BEGIN_EDIT
+        ], () => {
+          wrapper.find("a").simulate("click")
+        }, () => {
+          assert.equal(wrapper.find("input[text]").value, "")
```
Comment (noisecapella) [inline static/js/components/CreateCommentForm_test.js]: This isn't getting run, you need to put this in the `then` instead

**Comment by noisecapella** [comment]
It looks like the number of comments isn't getting updated after a comment is made:

screenshot from 2017-08-04 11-01-14

**Comment by rhysyngsun** [comment]
@noisecapella I've addressed your feedback and the bug you found

**Pull request merged**